plugins/content_distribution/lib/kContentDistributionManager.php

```diff
@@ -74,6 +74,19 @@ protected static function prepareDistributionJob(EntryDistribution $entryDistrib
 			$entryDistribution->getThumbAssetIds(),
 			$entryDistribution->getFlavorAssetIds()
 		)));
+
+		if($distributionProfile->shouldExcludeAudioFlavors())
+		{
+			$audios = assetPeer::retrieveAudioFlavorsByEntryID($entryDistribution->getEntryId());
+			$audioIds = array();
+			foreach($audios as $audio)
+			{
+				$id = $audio->getId();
+				if (in_array($id, $assetIds))
```

Comment:
i don't think you need to check that id is part of the assetIds
array_diff already returns the values that exist only in the first array